### DISCUSSION TITLE
lit3 demo using decorators?

### DISCUSSION THREAD

**Discussion by mbec-lbto** [lit/lit#4327] [Q&A]
Trying to upgrade a lit2 project to lit3, I am unable to get the decorators to work.
I went back to basic, using https://github.com/lit/lit-element-starter-js
- refactored it to use @customElement that seems to work
- refactored it to use @property that fails in some mysterious way (I can post the browser console output, and did put breakpoint on the browser debugger. I am not sure how to share that information).

babel.config.json
> {
> "assumptions": {
>     "setPublicClassFields": true
>   },
> "presets": ["@babel/preset-env"],
> "plugins": [
>     ["@babel/plugin-proposal-decorators", {"version": "2023-05"}],
>   ]
> }

could the `lit-element-starter-js` be expanded to show how that works?
Thank you

**Comment by augustjk** [comment]
We're working on updating the decorator section of the doc but we have some information on this in our Lit 3 migration doc here https://lit.dev/docs/releases/upgrade/#standard-decorator-migration

You're likely missing the `accessor` keyword in front of the decorated property.

You also don't need the
```json
{
  "assumptions": {
    "setPublicClassFields": true
  }
}
```
part of the babel config anymore.

Here's a minimal working example: https://stackblitz.com/edit/stackblitz-starters-ssueaq?file=index.js,babel.config.json

**Comment by mbec-lbto** [comment]
Thank you! yes, adding the `accessor` keyword fix it.
